src/main/scala/who/ate/all/the/pies/Mult.scala

```diff
@@ -0,0 +1,16 @@
+package who.ate.all.the.pies
+
+case class Mult(expressions: List[Expr]) extends Expr {
+  //  override def toString: String = expressions.map(_.toString).mkString(".")
+}
+
+case class MultSeq(elements: Expr*) extends Seq[Expr] with Expr {
```

Comment:
is this a better representation? I tried to go down this path but the pattern matching lex for sequences is awkward

Reply:
e.g.: ```
case MultSeq(x, PlusSeq(ps@_*), rest@_*) =>
        PlusSeq(ps.map(e => MultSeq(x :: e :: rest.toList:_*)):_*)
```

Reply:
`case MultSeq(x, PlusSeq(ps@_*), rest@_*) =>
        PlusSeq(ps.map(e => MultSeq(x :: e :: rest.toList:_*)):_*)`

Reply:
To me it's surprising that you want to be able to do Mult for more than two things, what makes you prefer e.g. Mult(2, 3, 4) over Mult(2, Mult(3, 4)). I think the code would be a lot easier to write for the latter

Reply:
But in answer to your original question, I think you definitely want something similar to this because it's nicer to write. Maybe the easiest way to get something would be to do like

```
object Mult {
  def apply(expressions: Expr*) = Mult(expressions.toList)
}
```

if you wanted to fit into what you've currently got

Reply:
so something like `case class Mult(expL: Expr,exprR: Expr) extends Expr`. It could be worth exploring yeah

Reply:
I suppose you start to build a binary tree then. Which would be more performant?

Reply:
The binary tree thing seems common in comp sci, maybe it has a bunch of algorithms designed for it.